### ISSUE TITLE
`react/prop-types` false positive with local variable `props` 

### ISSUE THREAD

**Issue by rhys-vdw** [yannickcr/eslint-plugin-react#1288]
Apparently a local variable called `props` will be incorrectly considered to be passed props in a functional component.

```js
function Foo() {
  const props = {}
  props.bar = 'bar'
  return <div {...props} />
}
```

```
warning  'bar' is missing in props validation               react/prop-types
```

Following is fine: 

```js
class Foo extends Component {
  render() {
    const props = {}
    props.bar = 'bar'
    return <div {...props} />
  }
}
```

**Comment by ljharb** [comment]
Worth fixing, but I have no idea why you'd ever want a variable called "props" in a React component :-)

**Comment by rhys-vdw** [comment]
@ljharb FYI

```
function GenericThankyou({ nextAssignmentId }) {
  const cardProps = {}
  if (nextAssignmentId == null) {
    cardProps.submitText = 'Back to the lobby'
    cardProps.onSubmit = navigateTo(RailsRoutes.lobby_path())
  } else {
    cardProps.submitText = 'Take the next test'
    cardProps.onSubmit = navigateTo(RailsRoutes.do_ordered_test_path(nextAssignmentId))
  }
  return (
    <TestStepCard {...cardProps}>
      <Title>{translate('test.complete.thank_you_1')}</Title>
      <p className='GenericThankyou-text'>
        {translate('test.complete.message_1')}
      </p>
    </TestStepCard>
  )
}
```

**Comment by ljharb** [comment]
(`cardProps` seems like the better name, since `props` looks like it's the GenericThankYou props 😉)

**Comment by rhys-vdw** [comment]
(agreed, but I decided to report it anyway)

**Comment by ljharb** [comment]
Fixed in #1365
